tracer/src/Datadog.Trace/AppSec/Waf/Encoder.cs

```diff
@@ -137,6 +137,7 @@ string GetItemsAsString()
                             var key = getKey(keyValue!);
                             if (string.IsNullOrEmpty(key))
                             {
+                                childrenCount--;
```

Comment:
For better maintainability, I feel like it would be better to have EnumerateItems return the number of items actually added, then have the caller set `childrenCount` accordingly (better separation of concerns). 
Can be left for a future refactoring though.

Reply:
Yes, agree, the design could be improved here.